src/presentation/api/controller/services.go

```diff
@@ -0,0 +1,248 @@
+package apiController
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+	voHelper "github.com/speedianet/os/src/domain/valueObject/helper"
+	internalDbInfra "github.com/speedianet/os/src/infra/internalDatabase"
+	apiHelper "github.com/speedianet/os/src/presentation/api/helper"
+	"github.com/speedianet/os/src/presentation/service"
+)
+
+type ServicesController struct {
+	serviceService *service.ServicesService
+}
+
+func NewServicesController(
+	persistentDbService *internalDbInfra.PersistentDatabaseService,
+) *ServicesController {
+	return &ServicesController{
+		serviceService: service.NewServicesService(persistentDbService),
+	}
+}
+
+// ReadServices	 godoc
+// @Summary      ReadServices
+// @Description  List installed services and their status.
+// @Tags         services
+// @Security     Bearer
+// @Accept       json
+// @Produce      json
+// @Success      200 {array} dto.InstalledServiceWithMetrics
+// @Router       /v1/services/ [get]
+func (controller *ServicesController) Read(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(c, controller.serviceService.Read())
+}
+
+// ReadInstallableServices	 godoc
+// @Summary      ReadInstallableServices
+// @Description  List installable services.
+// @Tags         services
+// @Security     Bearer
+// @Accept       json
+// @Produce      json
+// @Success      200 {array} entity.InstallableService
+// @Router       /v1/services/installables/ [get]
+func (controller *ServicesController) ReadInstallables(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(
+		c, controller.serviceService.ReadInstallables(),
+	)
+}
+
+func parseRawPortBindings(bindings []interface{}) []string {
+	rawPortBindings := []string{}
+	for _, rawPortBinding := range bindings {
+		rawPortBindingMap, assertOk := rawPortBinding.(map[string]interface{})
+		if !assertOk {
+			continue
+		}
+
+		rawPortStr, assertOk := rawPortBindingMap["port"].(string)
+		if !assertOk {
+			rawPortFloat, assertOk := rawPortBindingMap["port"].(float64)
+			if !assertOk {
+				continue
+			}
+			rawPortStr = strconv.FormatFloat(rawPortFloat, 'f', -1, 64)
+		}
```

Comment:
Why not use InterfaceToUint? Why is Port a float in here? Port should be uint16 since it only goes to 65535.

Reply:
For the type assertion, I had to use `float64` because `echo` itself was interpreting all numeric inputs as floats. Anyway, I used `InterfaceToString` instead of `InterfaceToUint` because I would need to convert to a string anyway, so I did that beforehand.